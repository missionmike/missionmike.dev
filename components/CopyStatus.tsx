import React from 'react';

interface ICopy {
  copied: boolean;
  copyText: string;
}

const CopyText = function ({ copied = false, copyText = '' }: ICopy) {
  let returnText = `(click to copy)`;

  if (copied === true) {
    returnText = `Copied!`;
  } else if (copyText && copyText === 'none') {
    returnText = ``;
  } else if (copyText) {
    returnText = copyText;
  }

  return <React.Fragment>{returnText}</React.Fragment>;
};

const CopyStatus = ({ copied = false, copyText = '' }: ICopy) => {
  return (
    <span className={`copy-status ${copied === true ? `copied text-pink-500` : ``}`}>
      <CopyText copied={copied} copyText={copyText} />
    </span>
  );
};

export default CopyStatus;
