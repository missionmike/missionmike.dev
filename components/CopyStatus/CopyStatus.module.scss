.copyStatus {
  font-style: italic;

  &[data-copied='true'] {
    color: blue;
  }
}
