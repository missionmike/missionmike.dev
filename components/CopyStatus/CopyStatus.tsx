import React from 'react';
import styles from './CopyStatus.module.scss';

const CopyStatus = ({
  copied = false,
  copyText = '(click to copy)',
}: {
  copied: boolean;
  copyText?: string;
}) => {
  return (
    <span className={styles.copyStatus} data-copied={copied}>
      <React.Fragment>{copied === true ? `Copied!` : copyText}</React.Fragment>
    </span>
  );
};

export { CopyStatus };
