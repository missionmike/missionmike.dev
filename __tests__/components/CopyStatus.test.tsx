import { CopyStatus } from 'components/CopyStatus/CopyStatus';
import React from 'react';
import { render } from '@testing-library/react';

describe('CopyStatus component', () => {
  it('renders the component with default copy text', () => {
    const { container } = render(<CopyStatus />);
    expect(container).toMatchSnapshot();
  });

  it('renders the component with custom copy text', () => {
    const { container } = render(<CopyStatus copyText="Click to copy!" />);
    expect(container).toMatchSnapshot();
  });
});
