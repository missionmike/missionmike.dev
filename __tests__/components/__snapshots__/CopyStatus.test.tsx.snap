// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyStatus component renders the component when copied 1`] = `
<div>
  <span
    class="copyStatus"
    data-copied="true"
  >
    Copied!
  </span>
</div>
`;

exports[`CopyStatus component renders the component with custom copy text 1`] = `
<div>
  <span
    class="copyStatus"
    data-copied="true"
  >
    Copied!
  </span>
</div>
`;

exports[`CopyStatus component renders the component with default copy text 1`] = `
<div>
  <span
    class="copyStatus"
    data-copied="true"
  >
    Copied!
  </span>
</div>
`;
